### PR TITLE
V0.14.1 bump

### PR DESCRIPTION
**Pull request by josh146**
* Merges the bugfix PRs in the [v0.14.1 milestone](https://github.com/PennyLaneAI/pennylane/milestone/7?closed=1) into a holding branch for the v0.14.1 release.

* Bumps the version number

* Updates the changelog

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,42 @@
...
+* Fixes a bug where `Hamiltonian` objects created with non-list arguments
+  raised an error for arithmetic operations.
+  [(#1082)](https://github.com/PennyLaneAI/pennylane/pull/1082)
+
+* Fixes a bug where `Hamiltonian` objects with no coefficients or operations
+  would return a faulty result when used with `ExpvalCost`.
+  [(#1082)](https://github.com/PennyLaneAI/pennylane/pull/1082)
+
+* Fixes a testing bug where tests that required JAX would fail if JAX was not installed.
+  The tests will now instead be skipped if JAX can not be imported.
```
Comment (trbromley) [inline .github/CHANGELOG.md:24]:
```suggestion
  The tests will now instead be skipped if JAX cannot be imported.
```

File: pennylane/tape/qnode.py [hunk 2]
```diff
@@ -865,6 +884,7 @@ def qfunc_decorator(func):
             interface=interface,
             diff_method=diff_method,
             mutable=mutable,
+            max_expansion=max_expansion,
```
Comment (trbromley) [inline pennylane/tape/qnode.py:887]: I know this has already been code reviewed, but for my understanding - we have added the `max_expansion` attribute but have we implemented the effects of changing `max_expansion`?

**Review by trbromley — APPROVED** [review]
Just to check, the `v0.14.1-bugfix` branch is currently level with the `v0.14.0` release?

File: pennylane/vqe/vqe.py [hunk 3]
```diff
@@ -486,6 +486,10 @@ def __init__(
         self._multiple_devices = isinstance(device, Sequence)
         """Bool: Records if multiple devices are input"""
 
+        if all(c == 0 for c in coeffs) or not coeffs:
+            self.cost_fn = lambda *args, **kwargs: np.array(0)
+            return
```
Comment (antalszava) [inline pennylane/vqe/vqe.py:491]: Thank you for relocating this! 🚀

File: tests/devices/test_default_qubit_jax.py [hunk 4]
```diff
@@ -475,6 +475,23 @@ def circuit(param):
...
+        @qml.qnode(dev, diff_method="backprop", interface="jax")
+        def circuit(param):
+            qml.RY(param, wires=0).inv()
+            return qml.expval(qml.PauliX(0))
+
+        x = 0.3
+        res = circuit(x)
+        assert np.allclose(res, -np.sin(x), atol=tol, rtol=0)
+
+        grad = jax.grad(lambda a: circuit(a).reshape(()))(x)
```
Comment (antalszava) [inline tests/devices/test_default_qubit_jax.py:492]:
```suggestion
        # Adjust grad func to be compatible when tested with both old and new cores of PennyLane
        grad = jax.grad(lambda a: circuit(a).reshape(()))(x)
```

**Review by antalszava — APPROVED** [review]
Looks good to me! 💯

File: pennylane/tape/qnode.py (hunk 2 above)
Comment (josh146) [reply pennylane/tape/qnode.py:887]: Yep! The change is on line 781, `max_expansion=10`.

**Pull request merged**